### ISSUE TITLE
How to make IodineGBA load the BIOS file directly from the server?

### ISSUE THREAD

**Issue by perelowski** [taisel/IodineGBA#41]
Hello,

I am trying to save some time when waiting for a game and skip the BIOS upload step. In order to do this I want to change the code to load the BIOS automatically upon the page load, from the same directory that IodineGBA is located. 

How to achieve that? I have checked the ROMLoadGlueCode.js and there is a function named 'downloadFile' - unfortunately I cannot get it to work. Any help would be appreciated!

**Comment by taisel** [comment]
Sorry I haven't gotten back sooner. I don't like the idea of having the base GUI of having a specific way to load ROMs from the server. Should be relegated to modified GUIs.

That being said, the BIOS can be technically saved client side after first load, via the same methodology as the saves system.

**Comment by Heath123** [comment]
Caan games load without a BIOS?

**Comment by Onimishra** [comment]
Nope, there is no BIOS emulation in Iodine. It is however possible to use the OpenSource BIOS developed by Normatt, which can be found [here](https://github.com/Nebuleon/ReGBA/tree/master/bios). It uses a different splash screen than the standard GBA one, but it works mostly the same.

**Comment by roguesaloon** [comment]
This never really got answered before. I am now trying to do something similar. I have a rom (which I developed) and the open source BIOS on my server, and I want to know what javascript needs changing to get them to load in automatically when the emulator is loaded, so the user can get straight into the game.

**Comment by roguesaloon** [comment]
Thanks to the slightly outdated code found in over here https://github.com/jsemu2/gba, I was able to get this working the way I wanted it.

In in CoreGlueCode.js add the following four functions

`function downloadBIOS() {
    downloadFile("../gba_bios.bin", registerBIOS);
}
function registerBIOS() {
    processDownload(this, attachBIOS);
    downloadROM(location.hash.substr(1));
}
function downloadROM(gamename) {
    downloadFile("../game.gba", registerROM);
}
function registerROM() {
    processDownload(this, attachROM);
    IodineGUI.Iodine.play();
}`

Then call downloadBIOS(); from window.onload function. The directory should be relative to where the emulator is located on the server.
